examples/plot_gp_function.py

```diff
@@ -35,32 +35,56 @@ def posterior(optimizer, x_obs, y_obs, grid):
         return mu, sigma
     
     fig = plt.figure(figsize=(16, 10))
-    steps = len(list(optimizers.values())[0].space)
+    #steps = len(list(optimizers.values())[0].space)
+    steps = len(optimizers[0].space)
   
     gs = gridspec.GridSpec(2, 1, height_ratios=[2, 2]) 
     axis = plt.subplot(gs[0])
     af = plt.subplot(gs[1])
     n = len(optimizers)
     colmap = plt.cm.get_cmap(name='rainbow', lut=n)
     
+    '''
     x_obs = {}  # dizionario vuoto
-    y_obs={}
+    y_obs = {}
     mu={}
     sigma={}
-    utility_function={}
     utility={}
-    for acq, optimizer in optimizers.items():  # acq è la stringa-chiave, optimizer è l'oggetto
-        x_obs[acq] = np.array([[res["params"]["x"]] for res in optimizer.res])
-        y_obs[acq]=  np.array([res["target"] for res in optimizer.res])
-        mu[acq], sigma[acq] = posterior(optimizer, x_obs[acq], y_obs[acq], x)
-        utility_function[acq] = UtilityFunction(kind=acq, kappa = params[acq]['kappa'], xi = params[acq]['xi'])
-        utility[acq] = utility_function[acq].utility(x, optimizer._gp, optimizer._space.target.max())
+    acq=['ucb', 'ei', 'poi']
+    count=0
+    for optimizer in optimizers:  # acq è la stringa-chiave, optimizer è l'oggetto
+        utility_function={}
+        x_obs[acq[count]] = np.array([[res["params"]["x"]] for res in optimizer.res])
+        y_obs[acq[count]] =  np.array([res["target"] for res in optimizer.res])
+        mu[acq[count]], sigma[acq[count]] = posterior(optimizer, x_obs[acq[count]], y_obs[acq[count]], x)
+        utility_function[acq[count]] = UtilityFunction(kind=acq[count], kappa = params[acq[count]]['kappa'], xi = params[acq[count]]['xi'])
+        utility[acq[count]] = utility_function[acq[count]].utility(x, optimizer._gp, optimizer._space.target.max())
+        count = count+1
+'''
+    
+    x_obs = []
+    y_obs = []
+    mu=[]
+    sigma=[]
+    utility=[]
+    
+    acq = ['ucb', 'ei', 'poi']
+    for i, optimizer in enumerate(optimizers): # enumerate applicata a iterable -> sia valore che indice
+        x_obs.append(np.array([[res["params"]["x"]] for res in optimizer.res]))
+        y_obs.append(np.array([res["target"] for res in optimizer.res]))
+        mu_temp, sigma_temp = posterior(optimizer, x_obs[i], y_obs[i], x)
```

Comment:
Un po' poco leggibile... Potreste mettere gli `np.array(...)` in una `x_obs_i` e `y_obs_i` temporanee, per poi usarle sia in `append()` che in `posterior()`.